packages/url/src/clean-for-slug.js

```diff
@@ -25,8 +25,12 @@ export function cleanForSlug( string ) {
 	}
 	return (
 		removeAccents( string )
+			// Convert &nbsp, &ndash, and &mdash to hyphens.
+			.replace( /(&nbsp;|&ndash;|&mdash;)/g, '-' )
 			// Convert each group of whitespace, periods, and forward slashes to a hyphen.
 			.replace( /[\s\./]+/g, '-' )
+			// Remove all HTML entities.
+			.replace( /&\S+?;/g, '' )
```

Comment:
I borrowed this from the `WordCounter` file - https://github.com/WordPress/wordpress-develop/blob/163836d89fecd506e3a2cf23182e14237738a4cf/src/js/_enqueues/wp/utils/word-count.js#L75.